addons/api/mirage/factories/host-set.js

```diff
@@ -27,11 +27,10 @@ export default factory.extend({
   },
   preferred_endpoints() {
     if (this.type === 'plugin') {
-      const enpointsAmount = faker.number.int({ min: 1, max: 5 });
       let result = [];
-      for (let i = 0; i < enpointsAmount; ++i) {
-        result.push(faker.internet.ip());
-      }
+
+      result.push(faker.internet.ip());
```

Comment:
I noticed a weird behavior when testing this, [the](https://github.com/hashicorp/boundary-ui/pull/1957/files#diff-4cbf41fe5ec71ecf02bccd70b53f754ffef7911f0c6f228b139ff350010abb08R176-R179) `promise.all` doesn't wait for all the buttons to be clicked so I had to use a single entry for this array here. But this only happens in the tests and is not an issue when I manually test the code.

Reply:
Nice catch! I understand there is a testing issue, if I have time today, I will try to understand it better and check if there is a solution to it.

I am a bit concern, if we want to test i.e removing all filters, having just 1 item in the list might not cover all the usage of the component.

Reply:
We shouldn't change the mirage data just to make a test pass, we should figure out a way to fix the test

Reply:
I took a look at this but I'm surprised this worked before or that we didn't have an issue previously. The `Promise.all` works fine but the underlying issue is all the promises are run at the same time (aka all the buttons are clicked at once) which causes a race condition because the remove function uses the index of the array to remove them so it might not remove the right one.

Using a `for...of` loop to go over the promises sequentially fixes the issue

Reply:
thanks for looking into it, that makes sense. I will try your suggestion.